segmund.py

```diff
@@ -83,40 +73,31 @@ def get_activities():
 # */
 @app.route('/exchange_token', methods=['GET'])
 def register_user():
-    state = request.args.get('state')
     auth_code = request.args.get('code')
-    scope = request.args.get('scope')
 
     user = strava_service.register_user(auth_code)
 
+    # TODO: remove this condition?  I don't think the function can return None
```

Comment:
Agree. 

It looks like register_user returns a model object and we'll likely always want to call save below? The registration process is idempotent so if a user does it again I think we get a new token and may want to persist. I haven't dug in to see if Strava is smart enough to issue new tokens and expire the old ones in this scenario.

Reply:
You're right that we should persist the new one if a user re-registers.  If the API generates new tokens, we should use those ones.